animation_nodes/data_structures/meshes/mesh_data.pyx

```diff
@@ -190,10 +214,14 @@ cdef class Mesh:
         return mesh
 
     def copyAttributes(self, Mesh source):
-        meshAttributes = self.attributes
-        sourceMeshAttributes = source.attributes
-        for name, value in sourceMeshAttributes.items():
-            meshAttributes[name] = value.copy()
+        sourceMaterialIndices = source.materialIndices
+        if sourceMaterialIndices is not None:
+            self.materialIndices = sourceMaterialIndices.copy()
+
+        for ((meshAttributes, _), (sourceMeshAttributes, _)) in zip(
+                self.getMeshAttributes(), source.getMeshAttributes()):
+            for name, value in sourceMeshAttributes.items():
+                meshAttributes[name] = value.copy()
```

Comment:
I would remove the `getMeshAttributes` and loop explicitly over each of the dictionaries.

Reply:
Omar, the `getAttributeDictionaries` method works for all dictionaries.